protocol/0083-RFPR-on_chain_referral_program.md

```diff
@@ -202,27 +219,53 @@ message UpdateReferralSet{
 
 If a referral set is currently designated as a team, a referrer should be able to "close" their team to any new members by setting the `closed` field to `True`. Note, closing a team is the same as closing a referral set and as such all `ApplyReferralCode` transactions applying the referral code associated with the closed referrals set should be rejected.
 
+Note, if a referrer updates the `allow_list` defining the parties which are allowed to join the team, the updated list is only used to validate attempts to join the team, i.e. if an existing member is removed from the list, they are not removed from the team.
+
 If a referral set is currently designated as a team, a party is able to effectively "disband" a team by updating their referral set and setting their `is_team` value to `False`. Note a team should only be "disbanded" and removed from leaderboards at the end of the current epoch after rewards have been distributed.
 
 ### Applying a referral code
 
 To apply a referral code and become a referee, a party must fulfil the following criteria:
 
 - party must not currently be a **referrer**
+- party must not currently be a **referee** associated with a valid referral set (Note: for a set to be valid, the sets referrer must be meeting the staking requirement.)
 
 To become a referee, a referee must submit a signed `ApplyReferralCode` transaction with the following fields:
 
 - `referral_code`: the referral code they wish to apply
+- `alias`: Optional string defining an alias which should be associated with the key applying the referral code and should appear on leaderboards in the keys place.
```

Comment:
Should the alias be unique ? Is there anything to prevent every party to be named _wen-moon_ ? Do we have restrictions in size, characters, etc ?

Reply:
Good points, for length a restriction of `20` characters works, for character restriction can we use the same as what we have for the team name?

For whether the alias has to be unique I am less sure as the benefit seems small to me as the alias is purely cosmetic for leaderboards. If the alias replaced keys in forums / governance views etc then yeah I think unique to prevent impersonation would be important.

Out of interest, how expensive would it to be check uniqueness? Is there any risk of DoS attacks?

Reply:
I agree, it all depends on the reach this alias will have.

It can be expensive on the long run, I guess. Because it means to keep track of all aliases in memory.

Reply:
Resolving as "alias"/"profiles" delayed to future release.